pkg/utils/nodeutils.go

```diff
@@ -209,6 +209,14 @@ func McfgPoolLabelMatches(nodeSelector map[string]string, pool *mcfgv1.MachineCo
 		return false
 	}
 	// TODO(jaosorior): Make this work with MatchExpression
```

Comment:
nit: I would move this down to line 220 so that it's closest to the original code it was written for.